crates/nu-command/src/system/run_external.rs

```diff
@@ -304,6 +310,26 @@ impl ExternalCommand {
                 let stderr_receiver = ChannelReceiver::new(stderr_rx);
                 let exit_code_receiver = ValueReceiver::new(exit_code_rx);
 
+                #[cfg(unix)]
+                {
+                    use signal_hook::low_level::signal_name;
+                    use std::os::unix::process::ExitStatusExt;
+                    use std::time::Duration;
+
+                    // Don't block here if there's no sender
+                    if let Ok(status) = exit_status_rx.recv_timeout(Duration::from_millis(100)) {
```

Comment:
Is this a performance concern? It seems like it might add 100ms to external command invocation.

Reply:
There are two situations which lead `exit_status_rx` block 100ms.
1. child.wait() return an error.
2.  A lot of output to be redirected from child process. Currently, stdout/stderr redirection use bounded channel, the sender may block.

Reply:
Thanks for the explanation.